pkg/api/controller.go

```diff
@@ -377,6 +384,24 @@ func (c *Controller) LinkPhysicalAddress(w http.ResponseWriter, r *http.Request,
 	writeTime := time.Now()
 	physicalAddress, addressType := normalizePhysicalAddress(repo.StorageNamespace, StringValue(body.Staging.PhysicalAddress))
 
+	// validate token
+	addressTokenCreationTime, err := c.PathProvider.ResolvePathTime(physicalAddress)
+	if err != nil {
+		writeError(w, r, http.StatusInternalServerError, err)
+		return
+	}
+	addressTokenExpiryTime := addressTokenCreationTime.Add(ref.AddressTokenTime)
+	if addressTokenExpiryTime.Before(time.Now()) {
+		c.handleAPIError(ctx, w, r, graveler.ErrAddressTokenExpired)
+		return
+	}
```

Comment:
It feels to me like this belongs in the `VerifyAddressToken`

Reply:
removed. thanks